### ISSUE TITLE
SameDiff: Missing RNN Ops for TF import: LSTMBlockCell, BlockLSTM, GRUBlockCell

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#7121]
These are not mapped for import currently. Not sure on the details yet, such whether we can use the existing RNN libnd4j ops - or whether we need new ops in c++.

**Comment by AlexDBlack** [comment]
Ok, so LSTMBlockCell first...

We have this: https://github.com/deeplearning4j/deeplearning4j/blob/master/libnd4j/include/ops/declarable/generic/recurrent/lstmCell.cpp
Which seems to be based on this:
https://github.com/tensorflow/tensorflow/blob/r1.12/tensorflow/python/ops/rnn_cell_impl.py#L810-L1045

There are a number of differences, however:
* Number of outputs (2 vs. 7 - "extra" outputs are to make it efficiently backpropable)
* Projection weights (LSTMBlockCell doesn't support them)
* Weight format (TF packs input and recurrent weights together, but breaks peephole connections out into 3 arrays)

In short, there's enough differences to make it infeasible to use the existing lstmCell implementation.
We will need a new implementation that follows the TF conventions here.
Here's a reference for backprop (haven't seen forward pass code yet, but it's similar to our existing implementation anyway)
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/rnn/ops/lstm_ops.cc

**Comment by AlexDBlack** [comment]
BlockLSTM: "This is equivalent to applying LSTMBlockCell in a loop"
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/rnn/ops/lstm_ops.cc#L179-L221

---
---

GRUBlockCell:
Details here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/rnn/ops/gru_ops.cc
Our existing implementation: https://github.com/deeplearning4j/deeplearning4j/blob/master/libnd4j/include/ops/declarable/generic/recurrent/gru.cpp

There's some clear differences our implementation and the TF implementation we want to be able to import:
- Weights and biases layout (TF splits some up)
- More outputs in TF (4 outputs - gate activations, as used in backprop).

Looking at this, I think it should be fairly straightforward to modify our existing GRU implementation to be suitable for import.
